apps/theming-designer/src/components/FabricPalette.tsx

```diff
@@ -29,113 +33,285 @@ const tableClassName = mergeStyles({
   }
 });
 
-const slotClassName = mergeStyles({
-  display: 'flex',
-  alignItems: 'center',
-  overflow: 'auto'
-});
-
-const fabricPaletteColorBox = mergeStyles({
-  width: 15,
-  height: 15,
-  display: 'inline-block',
-  left: 5,
-  top: 5,
-  border: '1px solid black',
-  flexShrink: 0
-});
-
-export const FabricPalette: React.StatelessComponent<IFabricPaletteProps> = (props: IFabricPaletteProps) => {
-  const fabricSlotWidget = (fabricSlot: FabricSlots): JSX.Element => {
-    const slotRule = props.themeRules![FabricSlots[fabricSlot]];
+export class FabricPalette extends React.Component<IFabricPaletteProps, {}> {
```

Comment:
Is there a specific reason why you changed this from a function component to a class?

Reply:
I tried a couple things and at some point I had to add state to this class but I don't need that anymore so changed it back to functional - thanks for catching this!